### ISSUE TITLE
Suggestions interface

### ISSUE THREAD

**Issue by svineet** [goavki/apertium-html-tools#55]
The suggestions work on the backend was merged into the suggestions branch in this PR: https://github.com/goavki/apertium-apy/pull/18

Overall, the suggestions route works and is almost production ready, but the following needs to be taken care of (feel free to add anything in the comments):
* The UI in html tools needs refinement. spectie and firespeaker had some ideas, we could discuss this here.
* Tests could be written for apy endpoints.
* The whole mechanism could use a lot of rigorous testing, there might be bugs.
* ReCAPTCHA API supports localizations and I was thinking of integrating that during the GCI task, but it slipped out of my mind, could be a great idea.

Current interface is at https://github.com/svineet/apertium-html-tools/tree/suggest-word and some of that work was merged into https://github.com/goavki/apertium-html-tools/tree/suggestions but my later improvements still have open PRs to the suggestions branch of html-tools. So we need to work on organizing that stuff.

**Comment by jonorthwash** [comment]
I merged the two open suggestions PRs (https://github.com/goavki/apertium-html-tools/pull/35, https://github.com/goavki/apertium-html-tools/pull/37) into the suggestions branch.  Now @Kira-D should go through your list of to-do points.

**Comment by Kira-D** [comment]
@svineet can you, please, give me more information about the UI refinement?
What sort of tests do you have in mind?

**Comment by sushain97** [comment]
moved from PR:

@shardulc @jonorthwash I went ahead and set it up on my server (with a recaptcha key) so we can easily test: https://azurite.skc.name/apertium-html-tools/build/index.eng.html

You need to check `Mark unknown words` for it to take effect (type some nonsense to translate and you'll see the red).

There are definitely some UI touches that need to take place but I haven't thoroughly tested it yet. Whoever can get to it first should take a crack at it!

I also have a hunch that it's making my CPU spike... Here be dragons. Would appreciate it if someone could confirm that it's not just me.

**Comment by shardulc** [comment]
@sushain97 When I click the suggest button, nothing happens and I cannot see any network activity either. (The captcha works fine.) However, I can't debug this because I only see `min.js` :smiley:

**Comment by sushain97** [comment]
Let me run make debug next time I'm not afk and it'll be easier.

**Comment by sushain97** [comment]
@shardulc https://azurite.skc.name/apertium-html-tools/build/index.debug.html

Don't try too much to debug it ;) Just make a list of problems.

**Comment by shardulc** [comment]
List of problems:
 1. `/suggest` is not yet implemented on APy, not even a dummy
 1. The captcha is off-center and the text inside the captcha overflows it (see image); perhaps it would be better if we used an [invisible captcha](https://developers.google.com/recaptcha/docs/invisible)?
 1. Might be because of point 1, but after you make the suggestion, the modal doesn't go away
 1. You can click the suggest button, and generate a POST to APy, even if the captcha is unfulfilled
 1. Do we want a lemma translation? Or an in-context translation which may be ambiguously inflected? Either way, the message should be more explicit.
 1. It's not immediately obvious that the red unknown words can be clicked.

![point 2](https://screenshots.firefoxusercontent.com/images/ea06bc9e-cb69-416d-9dbe-b9567fc5f956.png)

**Comment by sushain97** [comment]
Seems like a good set of issues to me, i.e. sufficiently specific to create a GCI task.

Did you not experience CPU spikes while typing in the source text on this branch?

**Comment by shardulc** [comment]
I tried 10 times and I experienced CPU spikes only once, on my first try. The other 9 times were completely normal CPU activity.
